x-pack/plugins/security_solution/server/lib/detection_engine/prebuilt_rules/logic/diff/calculation/calculate_rule_fields_diff.ts

```diff
@@ -47,63 +48,76 @@ const TARGET_TYPE_ERROR = `Target version can't be of different rule type`;
 export const calculateRuleFieldsDiff = (
   ruleVersions: ThreeVersionsOf<DiffableRule>
 ): RuleFieldsDiff => {
-  validateRuleVersions(ruleVersions);
-
   const commonFieldsDiff = calculateCommonFieldsDiff(ruleVersions);
   // eslint-disable-next-line @typescript-eslint/naming-convention
   const { base_version, current_version, target_version } = ruleVersions;
+  const hasBaseVersion = base_version !== MissingVersion;
 
   switch (current_version.type) {
     case 'query': {
-      invariant(base_version.type === 'query', BASE_TYPE_ERROR);
+      if (hasBaseVersion) {
+        invariant(base_version.type === 'query', BASE_TYPE_ERROR);
+      }
```

Comment:
nit: maybe extract these invariants to a common function.

Reply:
I tried to, but Typescript started complaining about the rule type mismatch. Unfortunately, I haven't found a clean way to fix that, so I left the checks as is.